### ISSUE TITLE
Spec fails in a time-dependent way

### ISSUE THREAD

**Issue by brunoparga** [tricycle/predictionbook#153]
`spec/models/score_calculator_spec.rb` fails in a time-dependent way. There are four tests (on lines 111, 134, 158 and 203) that were passing earlier today and are now failing, they might pass again come February 1.

I think the issue is related to the following:
```
# set up
[1] pry(main)> today = Time.zone.today
=> Thu, 31 Jan 2019
[2] pry(main)> halloween = today - 3.months
=> Wed, 31 Oct 2018

# [3] and [4] give the same date, as expected
[3] pry(main)> halloween + 1.month
=> Fri, 30 Nov 2018
[4] pry(main)> today - 2.months
=> Fri, 30 Nov 2018

# lines [5], [6] and [7] should all give the same result, but don't
[5] pry(main)> halloween + 1.month + 1.month
=> Sun, 30 Dec 2018
[6] pry(main)> halloween + 2.months
=> Mon, 31 Dec 2018
[7] pry(main)> today - 1.month
=> Mon, 31 Dec 2018

# same with [8] and [9]
[8] pry(main)> halloween + 1.month + 1.month + 1.month
=> Wed, 30 Jan 2019
[9] pry(main)> halloween + 3.months
=> Thu, 31 Jan 2019
```

So the issue seems to be that adding months piecemeal (as in [5] and [8]) gives a different result than adding an n-month chunk at a time (as in [6] and [9]). Maybe the only days in the year when this test would fail are today, Jan 31, and Aug 31, since they're ending two 31-day months in a row.

Does this need fixed? What should we do?

**Comment by michaelwebb76** [comment]
The best solution would be to start using something like [Timecop](https://github.com/travisjeffery/timecop) to freeze the time to a specific date/time before running any of the specs that are breaking.

**Comment by brunoparga** [comment]
Solved by PR [#154](https://github.com/tricycle/predictionbook/pull/154).

**Comment by brunoparga** [comment]
Solved by PR [#154](https://github.com/tricycle/predictionbook/pull/154).
